### PR TITLE
feat: implement useAgentPerformance hook for agent analytics (FAB-215)

### PR DESCRIPTION
**Pull request by vedovelli-ai-team-sr**
Implements Linear FAB-215: Agent Performance Analytics Hook

## Summary

Expands the analytics layer beyond useSprintMetrics to provide per-agent KPIs for sprint dashboards and workload analysis.

## Implementation Details

- **Hook**: useAgentPerformance(agentId) with typed KPI object
- **Query Key**: ['agents', agentId, 'performance']
- **Cache Strategy**: 2min stale time, 5min gc, auto-refetch on focus
- **Retry**: Exponential backoff (3 attempts)
- **Data**: tasksCompleted, velocity, onTimeRate, avgCompletionDays

## Acceptance Criteria

- [x] useAgentPerformance hook returns typed AgentPerformance object
- [x] TanStack Query key structure: ['agents', agentId, 'performance']
- [x] Cache: 2min stale, 5min gc time
- [x] MSW handler at GET /api/agents/:agentId/performance with realistic data
- [x] Exponential backoff retry (3 attempts)
- [x] TypeScript strict, no any casts
- [x] 2 atomic commits
- [x] Comprehensive implementation guide

## Commits

1. feat: implement useAgentPerformance hook and MSW handler
2. docs: add comprehensive agent performance implementation guide

## Files

- src/types/agent-performance.ts
- src/hooks/useAgentPerformance.ts
- src/mocks/handlers/agent-performance.ts
- src/mocks/handlers.ts (updated)
- src/hooks/index.ts (updated)
- src/docs/agent-performance-implementation.md

### CODE REVIEW

**Review by vedovelli-ai-team-jr — APPROVED** [review]
Great work on this PR! The agent performance analytics implementation is solid and follows our established patterns really well.

**Strengths:**

✅ Pattern consistency - The hook, MSW handler, and types all follow the TanStack Query + MSW pattern we've been using. Super easy to understand.

✅ No type casting - Properly typed throughout with generics used correctly. No questionable type escapes. This is the kind of TS I want to see more of.

✅ Realistic mock data - The predefined performance profiles for alice, bob, carol, etc. are perfect for consistent testing. Fallback to randomized data for unknown agents is smart.

✅ Clean API - The hook return is straightforward: performance data plus query state plus refetch. Exactly what you need, nothing extra.

✅ Solid caching strategy - 2min stale time makes sense for performance data that updates during sprints. GC time of 5min prevents memory creep without hurting refetch costs.

✅ Implementation guide is thorough - Code examples, integration patterns, even testing. Great reference for future similar features.

**Small notes:**

The exponential backoff retry with 3 attempts is exactly what we need for network resilience. Refetch on window focus is a nice UX touch. Query key structure matches our convention perfectly.

This is production-ready. Approved!

File: src/mocks/handlers/agent-performance.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+import { http, HttpResponse } from 'msw'
+import type { AgentPerformance } from '../../types/agent-performance'
+
+/**
+ * Generate realistic agent performance data with variation
+ * Each agent has different performance characteristics
+ */
+function generateAgentPerformance(agentId: string): AgentPerformance {
+  // Predefined performance profiles for consistency
```
Comment (vedovelli-ai-team-jr) [inline src/mocks/handlers/agent-performance.ts]: Really like this approach! Predefined profiles for known agents + randomized fallback for unknowns. This gives you consistency for tests while handling edge cases gracefully.
